EIPS/eip-7723.md

```diff
@@ -0,0 +1,80 @@
+---
+eip: 7723
+title: Network Upgrade Inclusion Stages
+description: Overview of the various stages Core EIPs go through before their activation in network upgrades.
+author: Tim Beiko (@timbeiko)
+discussions-to: https://ethereum-magicians.org/t/eip-7723-network-upgrade-inclusion-stages/20281
+status: Draft
+type: Meta
+created: 2024-06-12
+---
+
+## Abstract
+
+Define the stages that Core EIPs go through in the process of planning network upgrades: `Proposed for Inclusion`, `Considered for Inclusion`, `Scheduled for Inclusion`, and `Included`.
+
+## Motivation
+
+When planning network upgrades, EIPs go through the `Considered for Inclusion` and `Included` stages. However, definitions for these terms are either outdated or non-existent. 
+
+Additionally, there isn't a formalized way for the Ethereum community to signal which EIPs they'd like to see included in a network upgrade, or for implementation teams to keep track of these. 
+
+This EIP proposes definitions for each of the following stages: `Proposed for Inclusion`, `Considered for Inclusion`, `Scheduled for Inclusion`, and `Included`. It also provides context and guidelines around when and how EIPs should be moved from one stage to the next. 
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Context: Upgrade Meta EIPs
+
+When planning a network upgrade, an Upgrade Meta EIP **MUST** be drafted to list EIPs in various stages of consideration. This Meta EIP **SHOULD** include three categories in its specification section: `Proposed for Inclusion`, `Considered for Inclusion` and `Scheduled for Inclusion`. Even if a category is empty, it **SHOULD** be included in the initial draft for clarity. 
+
+When the Upgrade Meta EIP is moved to `Last Call`, the `Proposed for Inclusion` and `Considered for Inclusion` lists **SHOULD** be removed, leaving only `Scheduled for Inclusion`. 
+
+Before the Upgrade Meta EIP is moved to `Final`, the `Scheduled for Inclusion` stage **MUST** be renamed to `Included` and contain only EIPs that were activated in the upgrade. 
+
+### Proposed for Inclusion
+
+To formally propose the inclusion of a Core EIP in a network upgrade, someone **MUST** open a pull request against the Upgrade Meta EIP to add the EIP in the `Proposed for Inclusion` section. Reasonable pull requests **SHOULD** be merged in a timely fashion by the Upgrade Meta EIP author.  
+
+At this stage, implementation teams **SHOULD** review the EIP in the context of including it in the next upgrade. 
+
+Note that EIPs must be `Proposed for Inclusion` for each network upgrade. In other words, proposals do not "carry over" to the next upgrade if an EIP is not included in the one it was first proposed for. 
```

Comment:
Is it out of scope for this proposal to categorize a rolling way to track community interest in certain enhancements for the next upgrade? Cataloguing this on a continuous basis might help reduce the perceived scope creep and crunch around fork scheduling time, which is infrequent enough that it causes frustration with the process.

We really only do this for fork drivers like Verkle.

Reply:
Maybe could be solved as easily as keeping an N+1 Meta-EIP model for forks on the repo

Reply:
> Maybe could be solved as easily as keeping an N+1 Meta-EIP model for forks on the repo

@non-fungible-nelson Meta EIP for Osaka already exists
https://eips.ethereum.org/EIPS/eip-7607

Reply:
I'm not sure I'd want this as part of this EIP, but a simple rule could be:

Current fork: proposals can move up to SFI
Next fork: proposals can move up to CFI
Fork +2: things can be PFI'd

That said, my sense is most things in Fork+2 would end up stale by the time we actually get to it, so I prefer having only two at the time.